### PR TITLE
[JBIDE-23821] fix NPE when existing server has no type

### PR DESCRIPTION
**Pull request by adietish**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by adietish** [comment]
@robstryker +1 please?

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java [hunk 1]
```diff
@@ -91,15 +92,53 @@
...
 	
+	protected abstract class OpenShiftServerPredicate implements Predicate<IServer> {
+
+		@Override
+		public final boolean test(IServer server) {
+			return getServerType().equals(server.getServerType())
+					&& testServer(server);
+		}
+		
+		abstract protected boolean testServer(IServer server);
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java]: oh, a leftover of a try that I eventually stopped.

File: tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/core/server/OpenShiftServerUtilsTest.java [hunk 2]
```diff
@@ -246,4 +249,74 @@ public void should_not_show_jmx_on_non_java_project() throws CoreException {
 	    doReturn(eclipseProject.getName()).when(server).getAttribute(eq(OpenShiftServerUtils.ATTR_DEPLOYPROJECT), anyString());
 	    assertThat(OpenShiftServerUtils.isJavaProject(server)).isFalse();
 	}
+
+	@Test
+	public void shouldFindServerGivenListOfServersContainsServerWithNullType() throws CoreException {
+		// given
+		IServer os3Server = mockOS3Server("os3", "someService");
```
Comment (adietish) [inline tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/core/server/OpenShiftServerUtilsTest.java]: this is a test, there's no added value in defining a constant to make sure we're always using the very same string.

**Comment by adietish** [comment]
@robstryker +1 please?

**Comment by adietish** [comment]
@robstryker +1 please?

**Comment by robstryker** [comment]
+1

**Pull request merged**